subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/CodeQualityPluginConfigurationAttributesTest.groovy

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class CodeQualityPluginConfigurationAttributesTest extends AbstractIntegrationSpec {
+
+    def "plugin runtime configuration is configured as runtime classpath"() {
```

Comment:
I wrote this test in an attempt to better understand variant-aware resolution and to demonstrate the impact on resolution should we add attributes to the configurations of code quality tools.
Now I think that this test actually shows that configuring code quality tool configurations with runtime classpath attributes will effectively break multi-project builds with cross-project dependencies where code quality plugins are applied.
Because both the code quality tool configuration and the project configuration declare an outgoing variant with the same runtime classpath attributes and create an ambiguity during resolution.
Do I understand this correctly, and what would be the way forward?